syntax = "proto3";

package e8;

message PointwiseEstimatesRequest {
  repeated int64 user_ids = 1;
  repeated int64 content_ids = 2;
}

message PointwiseEstimatesResponse {
  // The probability that the users specified by the user_ids likes the pieces
  // of contents specified by the content_ids.
  repeated float probs = 1;
}

service PointwiseRanking {
  // Computes Pr(likes(Content[j]) | User[i], theta) in batch.
  rpc ComputePointwiseEstimates(PointwiseEstimatesRequest)
      returns (PointwiseEstimatesResponse);
}
